sncosmo/specdata.py

```diff
@@ -0,0 +1,347 @@
+    the size of ``bin_edges``.
+
+    Optionally, there is a single ``time`` associated with all of the
+    observations.
+
+    TODO: calibrated spectra? See spectrum.py where there is wave_unit and unit,
+    and everything is converted internally.
+    TODO: covariance?
+
+    Has attribute ``fluxcov`` which may be ``None``.
+
+    Parameters
+    ----------
+    TODO
+
+    """
+    def __init__(self, wave=None, flux=None, fluxerr=None, fluxcov=None,
+                 bin_edges=None, bin_starts=None, bin_ends=None, time=None):
+        # Extract the bin starts and ends
+        bin_starts, bin_ends = _parse_wavelength_information(
+            wave, bin_edges, bin_starts, bin_ends
+        )
+        self.bin_starts = bin_starts
+        self.bin_ends = bin_ends
+
+        # Make sure that the flux data matches up with the wavelength data.
+        self.flux = np.asarray(flux)
+        if not (len(self.bin_starts) == len(self.bin_ends) == len(self.flux)):
+            raise ValueError("unequal column lengths")
+
+        # Extract uncertainty information in whatever form it came in.
+        self._fluxerr = None
+        self._fluxcov = None
+
+        if fluxerr is not None:
+            if fluxcov is not None:
+                raise ValueError("can only specify one of fluxerr and fluxcov")
+            self._fluxerr = np.array(fluxerr)
+            if len(self._fluxerr) != len(self.bin_starts):
+                raise ValueError("unequal column lengths")
+        elif fluxcov is not None:
+            self._fluxcov = np.array(fluxcov)
+            if not (len(self.bin_starts) == self._fluxcov.shape[0] ==
+                    self._fluxcov.shape[1]):
+                raise ValueError("unequal column lengths")
+
+        self.time = time
+
+    @property
+    def wave(self):
+        """Return the centers of each bin."""
+        return (self.bin_starts + self.bin_ends) / 2.
+
+    @property
+    def fluxerr(self):
+        """Return the uncertainties on each flux bin"""
+        if self._fluxerr is not None:
+            return self._fluxerr
+        elif self._fluxcov is not None:
+            return np.sqrt(np.diag(self._fluxcov))
+        else:
+            raise ValueError("no uncertainty information available")
+
+    @property
+    def fluxcov(self):
+        """Return the covariance matrix"""
+        if self._fluxcov is not None:
+            return self._fluxcov
+        elif self._fluxerr is not None:
+            return np.diag(self._fluxerr**2)
+        else:
+            raise ValueError("no uncertainty information available")
+
+    def get_bands(self):
+        """Return a list of bandpass objects for each wavelength element."""
+
+        dwave = 0.00001
+        bands = []
+
+        for bin_start, bin_end in zip(self.bin_starts, self.bin_ends):
+            bands.append(Bandpass(
+                [bin_start, bin_start + dwave, bin_end, bin_end + dwave],
+                [0., 1., 1., 0.],
+            ))
```

Comment:
I *think* `Bandpass` has zero transmission outside the wavelength bounds, so I think you can simplify to
```python
Bandpass([bin_start, bin_end], [1.0, 1.0])
```